R/join_keys-extract.R

```diff
@@ -134,12 +138,13 @@
 #'
 #' # Setting a single relationship pair ---
 #'
-#' jk["ds4", "ds1"] <- c("pk4" = "pk1")
+#' jk["ds1", "ds4"] <- c("pk1" = "pk4")
 #'
 #' # Removing a key ---
 #'
 #' jk["ds5", "ds5"] <- NULL
-`[<-.join_keys` <- function(x, i, j, value) {
+`[<-.join_keys` <- function(x, i, j, parent = c("i", "j", "none"), value) {
+  parent <- match.arg(parent)
```

Comment:
`parent` have different default values and thus explaination of the same argument differs between `[<-.join_keys` and `join_key`

```suggestion
`[<-.join_keys` <- function(x, i, j, directed = TRUE, value) {
  checkmate::assert_flag(directed)
```

Reply:
This is a solid argument. The `match.arg` pattern was my preferred solution but I'm not convinced anymore.

Reply:
Pushed new version with `directed = TRUE` parameter.